src/ConnectedMode/ServerSentEvents/ServerSentEventSessionManager.cs

```diff
@@ -18,29 +18,109 @@
 
 namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
 {
-    internal interface IServerSentEventSessionManager
+    internal interface IServerSentEventSessionManager : IDisposable
     {
     }
 
     [Export(typeof(IServerSentEventSessionManager))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class ServerSentEventSessionManager : IServerSentEventSessionManager
+    internal sealed class ServerSentEventSessionManager : IServerSentEventSessionManager
     {
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
+        private readonly IIssueChangedServerEventSourcePublisher issueChangedServerEventSourcePublisher;
+        private readonly ISonarQubeService sonarQubeClient;
+        private readonly IServerSentEventPump eventPump;
+        private readonly ITaintServerEventSourcePublisher taintServerEventSourcePublisher;
+        private readonly IThreadHandling threadHandling;
+        private bool disposed;
+        private CancellationTokenSource sessionTokenSource;
+
         [ImportingConstructor]
         public ServerSentEventSessionManager(
             IActiveSolutionBoundTracker activeSolutionBoundTracker,
             ISonarQubeService sonarQubeClient,
+            IServerSentEventPump eventPump,
             IIssueChangedServerEventSourcePublisher issueChangedServerEventSourcePublisher,
-            ITaintServerEventSourcePublisher taintServerEventSourcePublisher)
+            ITaintServerEventSourcePublisher taintServerEventSourcePublisher,
+            IThreadHandling threadHandling)
+        {
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.sonarQubeClient = sonarQubeClient;
+            this.eventPump = eventPump;
+            this.issueChangedServerEventSourcePublisher = issueChangedServerEventSourcePublisher;
+            this.taintServerEventSourcePublisher = taintServerEventSourcePublisher;
+            this.threadHandling = threadHandling;
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionChanged;
+        }
+
+        public void Dispose()
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            activeSolutionBoundTracker.SolutionBindingChanged -= OnSolutionChanged;
+            sessionTokenSource?.Cancel();
+            issueChangedServerEventSourcePublisher?.Dispose();
+            taintServerEventSourcePublisher?.Dispose();
+            disposed = true;
+        }
+
+        internal /*for testing*/ void OnSolutionChanged(object sender, ActiveSolutionBindingEventArgs activeSolutionBindingEventArgs)
+        {
+            var bindingConfiguration = activeSolutionBindingEventArgs.Configuration;
+            var isOpen = !bindingConfiguration.Equals(BindingConfiguration.Standalone);
+
+            if (isOpen)
+            {
+                var projectKey = bindingConfiguration.Project.ProjectKey;
+                Connect(projectKey);
```

Comment:
Connect is slightly misleading, since there is a Connect method on sonarQubeClient but that's not the method we're calling. Perhaps `InitializeSession` and `EndSession` instead of Disconnect?